code/modules/chemistry/Chemistry-Machinery.dm

```diff
@@ -386,9 +390,10 @@
 			attack_hand(usr)
 			return
 		else if (href_list["eject"])
-			if (src.beaker)
-				beaker.set_loc(src.output_target)
-			usr.put_in_hand_or_eject(beaker) // try to eject it into the users hand, if we can
+			var/obj/item/I = src.beaker
+			if (I)
+				I.set_loc(src.output_target)
```

Comment:
Is there a particular reason you're casting to `var/obj/item/I` here and other places? Seems unnecessary?

Reply:
I added the variable because I believe as soon as the beaker is moved outside of the machine by `set_loc` or some other proc, the `Exited` proc immediately triggers and sets `src.beaker`/`src.inserted` to `null`, causing any procs below such as `put_in_hand_or_eject` to fail since the reference to the beaker is missing.

If you're talking about the `/obj/item/` type though, I can change that if you want.

Reply:
Ah gotcha. Does that mean some of the following code such as 
```
beaker = null
icon_state = "mixer0"
src.updateUsrDialog()
```
is redundant since it's already in `Exited`? If so, could you remove those redundant lines and leave a comment about how `Exited` is called upon `set_loc`?